src/domain/filtering/filter-flow.ts

```diff
@@ -108,5 +108,5 @@ export const filterFlowByEntry = (flow: Flow, filter: FilterEntry): boolean => {
     }
   }
 
-  return fromOk || toOk;
+  return filter.negative !== (fromOk || toOk);
```

Comment:
I guess similar logic should be applied for tcp flag filtering (Line 101 above). Could you please arrange that?